### PR TITLE
New framework for Electric Eye and its auditors

### PR DESCRIPTION
**Pull request by aclatham**
This PR adds a new controller for all auditors. This controller adds support for unit testing and different outputs for findings. The new supported outputs are csv, json, Security Hub, and DisruptOps. This required a new framework for the auditors and a new testing framework for developing auditors. The new auditor framework provides support for govcloud (no more separate govcloud auditors). There is also a new click cli to help with developing and running locally.

### CODE REVIEW

**Comment by jonrau1** [comment]
@aclatham @brandypeterson @jodybrazil for adding `Compliance.RelatedRequirements` would you rather me do it post-merge or provide lists to you to put into them in-line?

**Comment by jodybrazil** [comment]
Post-merge would be ideal as I think the PR is already big enough.  But, up
to you, either way works.  And, if you have the compliance control list, we
can update and submit another PR post-merge.

**Comment by jonrau1** [comment]
@jodybrazil gotcha. In that case I will lay it down with my own PR. I also have to update some other stuff like architecture & point V1 users to the perm branch

Will also likely need to touch base on getting GCP checks from "classic" into the new framework

File: eeauditor/auditors/aws/Amazon_EC2_Image_Builder_Auditor.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+            "Types": [
+                "Software and Configuration Checks/AWS Security Best Practices",
+                "Effects/Data Exposure",
+            ],
+            "FirstObservedAt": iso8601Time,
+            "CreatedAt": iso8601Time,
+            "UpdatedAt": iso8601Time,
+            "Severity": {"Label": "MEDIUM"},
+            "Confidence": 99,
+            "Title": "[imagebuilder.2] Image recipes should encrypt EBS volumes",
```
Comment (jonrau1) [inline eeauditor/auditors/aws/Amazon_EC2_Image_Builder_Auditor.py:189]: nit: should be [ImageBuilder.x] to match the case of existing auditors. (nit applies to all `Title` in the ASFF for this one)

File: eeauditor/auditors/aws/Amazon_CloudFront_Auditor.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                        "Recommendation": {
+                            "Text": "For more information on key pairs for CloudFront trusted signers refer to the Creating CloudFront Key Pairs for Your Trusted Signers section of the Amazon CloudFront Developer Guide",
+                            "Url": "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-trusted-signers.html#private-content-creating-cloudfront-key-pairs",
+                        }
+                    },
+                    "ProductFields": {"Product Name": "ElectricEye"},
+                    "Resources": [
+                        {
+                            "Type": "AwsCloudFrontDistribution",
+                            "Id": f"{awsPartition.upper()}::::Account:{awsAccountId}",
```
Comment (jonrau1) [inline eeauditor/auditors/aws/Amazon_CloudFront_Auditor.py:78]: When the `Resource.*.Id` is *not* `AwsAccount` the ID should be the ARN of the resource that was audited.

File: eeauditor/auditors/aws/Amazon_Kinesis_Analytics_Auditor.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                    "Recommendation": {
+                        "Text": "For more information on monitoring applications using CloudWatch Logs refer to the Best Practices for Kinesis Data Analytics for Apache Flink section of the Amazon Kinesis Data Analytics Developer Guide",
+                        "Url": "https://docs.aws.amazon.com/kinesisanalytics/latest/java/best-practices.html#how-dev-bp-logging",
+                    }
+                },
+                "ProductFields": {"Product Name": "ElectricEye"},
+                "Resources": [
+                    {
+                        "Type": "AwsKinesisDataAnalyticsApplication",
+                        "Id": f"{awsPartition.upper()}::::Account:{awsAccountId}",
```
Comment (jonrau1) [inline eeauditor/auditors/aws/Amazon_Kinesis_Analytics_Auditor.py:79]: When the `Resource.*.Id` is *not* `AwsAccount` the ID should be the ARN of the resource that was audited. It looks like this is across all of the recently pushed Auditors - I won't keep putting these as to not spam everyone.

**Comment by jonrau1** [comment]
It looks like there are some ResourceId mismatches - the Account ID is provided instead of the actual Resource ARN. I highlighted it in 2 of the commited auditors but I have a feeling all of the new ones from #36 are like that.

**Comment by jonrau1** [comment]
Waiting for write access to your fork so I can update compliance & architecture diagrams. After that will be ready for merge.

**Pull request merged**